File: Endpoint.Utilities/Models/Vehicle.cs
using System.Collections.ObjectModel;
using System.ComponentModel.DataAnnotations.Schema;

namespace Endpoint.Utilities.Models;

[Table("Vehicles")]
public class Vehicle : BaseModel
{
    public bool IsRegistered { get; set; }
    public DateTime? LastUpdated { get; set; }

    // Complex type. Contact bliver ikke oprettet som egentabel. Men som en egenskab i vehicle
    public Contact Contact { get; set; }

    // Model reference 1 - M
    public Guid? ModelId { get; set; }
    public Model? Model { get; set; }
    // Features reference M - M
    public ICollection<Features>? Features { get; set; }

    /// <summary>
    /// Undgår nullreference
    /// </summary>
    public Vehicle()
    {
        Features = new Collection<Features>();
    }
}

File: Client/src/app/Models/vehicle.ts
import { Contact } from './contact';
import { KeyValuePair } from "./keyValuePair";

export interface Vehicle{
    id: string;
    model: KeyValuePair;
    make: KeyValuePair;
    isRegistered: boolean;
    features: KeyValuePair[];
    contact: Contact;
    lastUpdate: string;
}
